pkg/stanza/fileconsumer/internal/checkpoint/checkpoint.go

```diff
@@ -68,7 +82,9 @@ func Load(ctx context.Context, persister operator.Persister) ([]*reader.Metadata
 		if err = dec.Decode(rmd); err != nil {
 			return nil, err
 		}
-
+		if rmd.Fingerprint.HashBytes == 0 && rmd.Fingerprint.BytesUsed == 0 {
+			rmd.Fingerprint.AddHash()
+		}
```

Comment:
Based on my previous comment, I think we probably need to initialize the hash regardless. Actually, I wonder if there's any real benefit to saving and loading `HashBytes` and `BytesUsed` if these can both be derived from `FirstBytes`. Should _none_ of the new fields be exported?